diff --git a/dlls/ntdll/thread.c b/dlls/ntdll/thread.c
index 0962e70..91dec11 100644
--- a/dlls/ntdll/thread.c
+++ b/dlls/ntdll/thread.c
@@ -1224,7 +1224,7 @@ NTSTATUS get_thread_context( HANDLE handle, context_t *context, unsigned int fla
 
     if (ret == STATUS_PENDING)
     {
-        for (i = 0; i < 100; i++)
+        for (i = 0; i < 1000; i++)
         {
             SERVER_START_REQ( get_thread_context )
             {
