--- a/dlls/ddraw/device.c	
+++ a/dlls/ddraw/device.c	
@@ -4820,9 +4820,10 @@ static HRESULT d3d_device7_SetTexture(IDirect3DDevice7 *iface,
     struct ddraw_surface *surf = unsafe_impl_from_IDirectDrawSurface7(texture);
     struct wined3d_texture *wined3d_texture = NULL;
 
-    TRACE("iface %p, stage %u, texture %p.\n", iface, stage, texture);
+    TRACE("iface %p, stage %u, texture %p, surf %p, surf->surface_desc.ddsCaps.dwCaps %#x.\n",
+            iface, stage, texture, surf, surf ? surf->surface_desc.ddsCaps.dwCaps : 0);
 
-    if (surf && (surf->surface_desc.ddsCaps.dwCaps & DDSCAPS_TEXTURE))
+    if (surf /*&& (surf->surface_desc.ddsCaps.dwCaps & DDSCAPS_TEXTURE) */)
         wined3d_texture = surf->wined3d_texture;
 
     wined3d_mutex_lock();
diff --git a/dlls/ddraw/device.c b/dlls/ddraw/device.c
index 713d46616a..8e049e19d4 100644
--- a/dlls/ddraw/device.c
+++ b/dlls/ddraw/device.c
@@ -3920,8 +3920,16 @@ static HRESULT WINAPI d3d_device2_SetClipStatus(IDirect3DDevice2 *iface, D3DCLIP
  *****************************************************************************/
 static HRESULT WINAPI d3d_device7_GetClipStatus(IDirect3DDevice7 *iface, D3DCLIPSTATUS *clip_status)
 {
-    FIXME("iface %p, clip_status %p stub!\n", iface, clip_status);
-
+    FIXME("iface %p, clip_status %p, dwFlags %#x stub!\n", iface, clip_status, clip_status->dwFlags);
+
+    clip_status->minx = 0.0f;
+    clip_status->maxx = 4096.0f;
+    clip_status->miny = 0.0f;
+    clip_status->maxy = 4096.0f;
+    clip_status->minz = 0.0f;
+    clip_status->maxz = 0.0f;
+    clip_status->dwFlags = D3DCLIPSTATUS_EXTENTS2;
+    clip_status->dwStatus = 0;
     return D3D_OK;
 }
 
 
