diff -Naur a/dlls/kernelbase/kernelbase.spec b/dlls/kernelbase/kernelbase.spec
--- a/dlls/kernelbase/kernelbase.spec	2018-11-01 08:46:02.044609735 +0100
+++ b/dlls/kernelbase/kernelbase.spec	2018-11-01 08:50:33.537590337 +0100
@@ -1696,7 +1696,7 @@
 @ stdcall WaitForThreadpoolWorkCallbacks(ptr long) kernel32.WaitForThreadpoolWorkCallbacks
 # @ stub WaitForUserPolicyForegroundProcessingInternal
 @ stdcall WaitNamedPipeW(wstr long) kernel32.WaitNamedPipeW
-@ stdcall WaitOnAddress(ptr ptr long long)
+# @ stub WaitOnAddress
 @ stdcall WakeAllConditionVariable(ptr) kernel32.WakeAllConditionVariable
 @ stdcall WakeByAddressAll(ptr) ntdll.RtlWakeAddressAll
 @ stdcall WakeByAddressSingle(ptr) ntdll.RtlWakeAddressSingle
 
