--- a/dlls/dsound/dsound.c
+++ a/dlls/dsound/dsound.c
@@ -196,7 +196,9 @@ static ULONG DirectSoundDevice_AddRef(DirectSoundDevice * device)
 static ULONG DirectSoundDevice_Release(DirectSoundDevice * device)
 {
     HRESULT hr;
-    ULONG ref = InterlockedDecrement(&(device->ref));
+    ULONG ref;
+    Sleep(100);
+    ref = InterlockedDecrement(&(device->ref));
     TRACE("(%p) ref was %u\n", device, ref + 1);
     if (!ref) {
         int i;

--- a/dlls/wined3d/glsl_shader.c
+++ a/dlls/wined3d/glsl_shader.c
@@ -6709,6 +6709,12 @@ static void shader_glsl_setup_sm3_rasterizer_input(struct shader_glsl_priv *priv
             shader_addline(buffer, "gl_Layer = floatBitsToInt(outputs[%u])%s;\n",
                     output->register_idx, reg_mask);
         }
+        else if (output->sysval_semantic == WINED3D_SV_CLIP_DISTANCE)
+        {
+            gl_info->gl_ops.gl.p_glEnable(GL_CLIP_DISTANCE0+semantic_idx);
+            shader_addline(buffer, "gl_ClipDistance[%u] = outputs[%u]%s;\n",
+                    semantic_idx, output->register_idx, reg_mask);
+        }
         else if (output->sysval_semantic)
         {
             FIXME("Unhandled sysval semantic %#x.\n", output->sysval_semantic);
 
