From: Tk-Glitch <ti3nou@gmail.com>
Date: Fri, 12 Oct 2018 19:55:44 +0200
Subject: Halo online hack to fix black vertices on negative Z axis.
 https://bugs.winehq.org/show_bug.cgi?id=45375


diff --git a/dlls/wined3d/glsl_shader.c b/dlls/wined3d/glsl_shader.c
index d7fdd01..872426d 100644
--- a/dlls/wined3d/glsl_shader.c
+++ b/dlls/wined3d/glsl_shader.c
@@ -4164,8 +4164,15 @@ static void shader_glsl_scalar_op(const struct wined3d_shader_instruction *ins)
 
     if (dst_size > 1 && shader_version < WINED3D_SHADER_VERSION(4, 0))
         shader_addline(buffer, "vec%u(%s%s%s));\n", dst_size, prefix, src0_param.param_str, suffix);
-    else
+    else {
         shader_addline(buffer, "%s%s%s);\n", prefix, src0_param.param_str, suffix);
+        if (ins->handler_idx == WINED3DSIH_RCP) {
+            struct glsl_dst_param dst_param;
+            shader_glsl_add_dst_param(ins, &ins->dst[0], &dst_param);
+            shader_addline(buffer, "%s%s = ((isinf(%s%s)) ? 0.0 : %s%s);\n", dst_param.reg_name, dst_param.mask_str,
+                    dst_param.reg_name, dst_param.mask_str, dst_param.reg_name, dst_param.mask_str);
+        }
+    }
 }
 
 /** Process the WINED3DSIO_EXPP instruction in GLSL:
