From bf17687545f78f5201a0e934589f91657883e10d Mon Sep 17 00:00:00 2001
From: Tk-Glitch <ti3nou@gmail.com>
Date: Wed, 3 Apr 2019 20:57:29 +0200
Subject: proton all prebuilt


diff --git a/proton b/proton
index b49f3ee01..39092c1d2 100755
--- a/proton
+++ b/proton
@@ -365,7 +366,7 @@ with prefix_lock:
                 os.remove(dstfile)
             try_copy(steamdir + "/legacycompat/" + f, dstfile)
 
-    dxvkfiles = ("d3d11", "d3d10", "d3d10core", "d3d10_1", "dxgi")
+    dxvkfiles = ("d3d11", "d3d10", "d3d10core", "d3d10_1", "dxgi", "d3d9")
     def make_dxvk_links(dll_dir, link_dir):
         for f in dxvkfiles:
             dst = link_dir + "/" + f + ".dll"
@@ -376,22 +377,22 @@ with prefix_lock:
 
     if "wined3d" in config_opts:
         if not os.path.exists(prefix + "drive_c/windows/syswow64"):
-            #use gl-based wined3d for d3d11 and d3d10 32
+            #use gl-based wined3d for d3d11, d3d10 and d3d9 32
             make_dxvk_links(basedir + "/dist/lib/wine/fakedlls/",
                 prefix + "drive_c/windows/system32")
         else:
-            #use gl-based wined3d for d3d11 and d3d10 32+64
+            #use gl-based wined3d for d3d11, d3d10 and d3d9 32+64
             make_dxvk_links(basedir + "/dist/lib64/wine/fakedlls/",
                 prefix + "drive_c/windows/system32")
             make_dxvk_links(basedir + "/dist/lib/wine/fakedlls/",
                 prefix + "drive_c/windows/syswow64")
     else:
         if not os.path.exists(prefix + "drive_c/windows/syswow64"):
-            #use vulkan-based dxvk for d3d11 and d3d10 32
+            #use vulkan-based dxvk for d3d11 and d3d10 and d9vk for d3d9 32
             make_dxvk_links(basedir + "/dist/lib/wine/dxvk/",
                 prefix + "drive_c/windows/system32")
         else:
-            #use vulkan-based dxvk for d3d11 and d3d10 32+64
+            #use vulkan-based dxvk for d3d11 and d3d10 and d9vk for d3d9 32+64
             make_dxvk_links(basedir + "/dist/lib64/wine/dxvk/",
                 prefix + "drive_c/windows/system32")
             make_dxvk_links(basedir + "/dist/lib/wine/dxvk/",
 
