From 2c5529a6b40d87915681bdb3b60e771f92f8ec9d Mon Sep 17 00:00:00 2001
From: Tk-Glitch <ti3nou@gmail.com>
Date: Wed, 3 Apr 2019 21:05:38 +0200
Subject: proton all winelib


diff --git a/proton b/proton
index b49f3ee01..93b6792b8 100755
--- a/proton
+++ b/proton
@@ -365,40 +365,6 @@ with prefix_lock:
                 os.remove(dstfile)
             try_copy(steamdir + "/legacycompat/" + f, dstfile)
 
-    dxvkfiles = ("d3d11", "d3d10", "d3d10core", "d3d10_1", "dxgi")
-    def make_dxvk_links(dll_dir, link_dir):
-        for f in dxvkfiles:
-            dst = link_dir + "/" + f + ".dll"
-            src = dll_dir + "/" + f + ".dll"
-            if os.path.lexists(dst):
-                os.remove(dst)
-            os.symlink(src, dst)
-
-    if "wined3d" in config_opts:
-        if not os.path.exists(prefix + "drive_c/windows/syswow64"):
-            #use gl-based wined3d for d3d11 and d3d10 32
-            make_dxvk_links(basedir + "/dist/lib/wine/fakedlls/",
-                prefix + "drive_c/windows/system32")
-        else:
-            #use gl-based wined3d for d3d11 and d3d10 32+64
-            make_dxvk_links(basedir + "/dist/lib64/wine/fakedlls/",
-                prefix + "drive_c/windows/system32")
-            make_dxvk_links(basedir + "/dist/lib/wine/fakedlls/",
-                prefix + "drive_c/windows/syswow64")
-    else:
-        if not os.path.exists(prefix + "drive_c/windows/syswow64"):
-            #use vulkan-based dxvk for d3d11 and d3d10 32
-            make_dxvk_links(basedir + "/dist/lib/wine/dxvk/",
-                prefix + "drive_c/windows/system32")
-        else:
-            #use vulkan-based dxvk for d3d11 and d3d10 32+64
-            make_dxvk_links(basedir + "/dist/lib64/wine/dxvk/",
-                prefix + "drive_c/windows/system32")
-            make_dxvk_links(basedir + "/dist/lib/wine/dxvk/",
-                prefix + "drive_c/windows/syswow64")
-        for f in dxvkfiles:
-            dlloverrides[f] = "n"
-
 if "nod3d11" in config_opts:
     dlloverrides["d3d11"] = ""
     if "dxgi" in dlloverrides:
