diff --git a/proton b/proton
index 96b1b7e..998f94a 100755
--- a/proton
+++ b/proton
@@ -191,6 +191,7 @@ with dist_lock:
 
 env = dict(os.environ)
 dlloverrides = {
+        "steam.exe": "b", #always use our special built-in steam.exe
         "mfplay": "n" #disable built-in mfplay
     }
 
@@ -705,7 +705,7 @@ def dump_dbg_scripts():
                 f.write(" \"" + arg + "\"")
         f.write(")\n")
         dump_dbg_env(f)
-        f.write("\t\"" + wine_path + "\" \"${@:-${DEF_CMD[@]}}\"\n")
+        f.write("\t\"" + wine_path + "\" steam.exe \"${@:-${DEF_CMD[@]}}\"\n")
     os.chmod(tmpdir + "run", 0o755)
 
 def run():
@@ -714,7 +714,7 @@ def run():
             dump_dbg_scripts()
         except OSError:
             log("Unable to write debug scripts! " + str(sys.exc_info()[1]))
-    run_wine([wine_path] + sys.argv[2:])
+    run_wine([wine_path, "steam"] + sys.argv[2:])
 
 if sys.version_info[0] == 2:
     binary_stdout = sys.stdout
