From 72415470516f2d248e52203b60c6ee7c85d6e206 Mon Sep 17 00:00:00 2001
From: Tk-Glitch <ti3nou@gmail.com>
Date: Sat, 25 May 2019 19:52:10 +0200
Subject: wrong


diff --git a/lsteamclient/steamclient_manual_144.cpp b/lsteamclient/steamclient_manual_144.cpp
index e17dd76..d5f7651 100644
--- a/lsteamclient/steamclient_manual_144.cpp
+++ b/lsteamclient/steamclient_manual_144.cpp
@@ -42,7 +42,7 @@ void CDECL win_Release(struct winSteamNetworkingMessage_t_144 *win_msg)
 
 void lin_FreeData(struct SteamNetworkingMessage_t *lin_msg)
 {
-    struct msg_wrapper *msg = CONTAINING_RECORD(lin_msg, struct msg_wrapper, lin_msg);
+    struct msg_wrapper *msg = CONTAINING_RECORD(&lin_msg, struct msg_wrapper, lin_msg);
     if(msg->win_msg.m_pfnFreeData)
         ((void (*)(struct winSteamNetworkingMessage_t_144 *))msg->win_msg.m_pfnFreeData)(&msg->win_msg);
 }
